File: javascript/re-capture.js
const re = /(a(?:bc)+?)[de]$/;
const str = "regexp123abcbcd";
const repeatCount = 10000000;

const start = Date.now();
for (let i = 0; i < repeatCount; ++i) {
  re.exec(str);
}
const end = Date.now();

console.log(end - start);

File: lua/re_capture.lua
local re = "(a(?:bc)+?)[de]$"
local str = "regexp123abcbcd"
local repeatCount = 1000000

local start = os.clock()
for i = 1, repeatCount do
  str:match(re)
end
local _end = os.clock()

print((_end - start) * 1000)
